tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContentStore.scala

```diff
@@ -0,0 +1,50 @@
+package com.linagora.tmail.encrypted
+
+import org.apache.james.blob.api.BlobId
+import org.apache.james.mailbox.model.MessageId
+import org.reactivestreams.Publisher
+
+object EncryptedEmailContentStore {
+  val POSITION_NUMBER_START_AT: Int = 0
+}
+
+trait EncryptedEmailContentStore {
+  def store(messageId: MessageId, encryptedEmailContent: EncryptedEmailContent): Publisher[Unit]
+
+  def delete(messageId: MessageId): Publisher[Unit]
+
+  def retrieveFastView(messageId: MessageId): Publisher[EncryptedEmailFastView]
+
+  def retrieveDetailedView(messageId: MessageId): Publisher[EncryptedEmailDetailedView]
+
+  def retrieveAttachmentContent(messageId: MessageId, position: Int): Publisher[BlobId]
+}
+
+case class MessageNotFoundException(messageId: MessageId) extends RuntimeException
+
+case class AttachmentNotFoundException(messageId: MessageId, position: Int) extends RuntimeException
+
+object EncryptedEmailFastView {
+  def from(encryptedEmailContent: EncryptedEmailContent): EncryptedEmailFastView =
+    EncryptedEmailFastView(
+      encryptedPreview = encryptedEmailContent.encryptedPreview,
+      hasAttachment = encryptedEmailContent.hasAttachment)
+}
+
+case class EncryptedEmailFastView(encryptedPreview: String,
+                                  hasAttachment: Boolean)
+
+object EncryptedEmailDetailedView {
+  def from(encryptedEmailContent: EncryptedEmailContent): EncryptedEmailDetailedView =
+    EncryptedEmailDetailedView(
+      encryptedPreview = encryptedEmailContent.encryptedPreview,
+      encryptedHtml = encryptedEmailContent.encryptedHtml,
+      hasAttachment = encryptedEmailContent.hasAttachment,
+      encryptedAttachmentMetadata = encryptedEmailContent.encryptedAttachmentMetadata)
+}
+
+case class EncryptedEmailDetailedView(encryptedPreview: String,
+                                      encryptedHtml: String,
+                                      hasAttachment: Boolean,
+                                      encryptedAttachmentMetadata: Option[String])
```

Comment:
Shouldn't we strong type these strings?

Reply:
Your mean is 
`case class EncryptedAttachmentMetadata(value: Option[String]) extends AnyVal` ? 
and StrongType for all properties? 
Sometimes, I thought it is a bit verbose

Reply:
an extra line of code is verbose? :) strong typing is what we do in most of our jmap scala code though